corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -224,13 +225,49 @@ def form_exists(form_id, domain=None):
             return result.form_exists
 
     @staticmethod
-    @transaction.atomic
```

Comment:
Removed it from here since there is only one SQL call so atomicity is implicit. 

We could add it to the `hard_delete_case_and_forms` function which is where this get's called from. Atomicity isn't respected in a sharded environment though since transactions only apply to the proxy DB.